smack-integration-test/src/main/java/org/jivesoftware/smack/ChatTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ *
+ * Copyright 2015 Florian Schmaus
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jivesoftware.smack;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.jivesoftware.smackx.jiveproperties.JivePropertiesManager.addProperty;
+import static org.jivesoftware.smackx.jiveproperties.JivePropertiesManager.getProperty;
+
+import java.util.Date;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.chat.Chat;
+import org.jivesoftware.smack.chat.ChatManager;
+import org.jivesoftware.smack.filter.ThreadFilter;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.jiveproperties.JivePropertiesManager;
+import org.jxmpp.jid.impl.JidCreate;
+import org.jxmpp.stringprep.XmppStringprepException;
+
+/**
+ * @author Stawicki Adam
+ */
+public class ChatTest extends AbstractSmackIntegrationTest {
+
+    private final ChatManager chatManagerOne;
+
+    public ChatTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+        chatManagerOne = ChatManager.getInstanceFor(conOne);
+    }
+
+    @SmackIntegrationTest
+    public void testProperties() throws XmppStringprepException, NotConnectedException, Exception {
+        Chat newChat = chatManagerOne.createChat(conTwo.getUser());
+        PacketCollector collector = conTwo.createPacketCollector(new ThreadFilter(newChat.getThreadID()));
+        JivePropertiesManager.setJavaObjectEnabled(true);
```

Comment:
Tests should leave the state unchanged, so if you set this to true, then you need to reset it at the end. Create two methods `setUp()` and `tearDown()`, annoate them with @BeforeClass and @AfterClass, to set and reset the config of JivePropertiesManager.